map/src/map/components/layers/PlanRouteLayer.js

```diff
@@ -0,0 +1,153 @@
+
+
+    useEffect(() => {
+        if (ctx.createNewTrack) {
+            if (ctx.currentlyEditTrack) {
+                deleteOldRoute(map);
+                ctx.currentlyEditTrackDispatch({
+                    type: 'clean',
+                })
+            }
+            ctx.currentlyEditTrackDispatch({
+                type: 'createTrack',
+            })
+            setOpenPanelButtons(true);
+        }
+    }, [ctx.createNewTrack, ctx.setCreateNewTrack]);
+
+    function deleteOldRoute(map) {
+        let layersWithPolyline = [];
+        map._layers && Object.keys(map._layers).forEach(e => {
+                if (map._layers[e].planroute) {
+                    layersWithPolyline.push(map._layers[e]);
+                }
+            }
+        );
+        layersWithPolyline.forEach(function (item) {
+            map.removeLayer(item);
+        });
+    }
+
+    useEffect(() => {
+        if (ctx.currentlyEditTrack) {
+            if (ctx.currentlyEditTrack.startDraw) {
+                if (!ctx.currentlyEditTrack.newRouteLayer) {
+                    ctx.currentlyEditTrackDispatch({
+                        type: 'start',
+                        newRouteLayer: map.editTools.startPolyline(),
+                    })
+                    ctx.setSelectedGpxFile(null);
+                    ctx.setWeatherPoint(null);
+                }
+            }
+
+            if (ctx.currentlyEditTrack.deleteTrack) {
+                let selectedTrack = ctx.createdTracks.find(t => t.selected === true);
+                if (selectedTrack) {
+                    ctx.createdTracks.splice(ctx.createdTracks.indexOf(selectedTrack), 1);
+                    ctx.setCreatedTracks([...ctx.createdTracks]);
+                    if (ctx.currentlyEditTrack.newRouteLayer && map.hasLayer(ctx.currentlyEditTrack.newRouteLayer)) {
+                        map.removeLayer(ctx.currentlyEditTrack.newRouteLayer);
+                        ctx.currentlyEditTrackDispatch({
+                            type: 'delete',
+                        })
+                    }
+                }
+            }
+
+            //check delete point
+            if (ctx.currentlyEditTrack.deletePoint && ctx.currentlyEditTrack.deletePoint !== -1) {
+
+                ctx.currentlyEditTrack.pointsList.splice(ctx.currentlyEditTrack.deletePoint, 1);
+                deleteOldRoute(map);
+                ctx.currentlyEditTrack.pointsList = Utils.getPointsDist(ctx.currentlyEditTrack.pointsList);
+                ctx.currentlyEditTrack.newRouteLayer = map.editTools.addPolylineByPoints(ctx.currentlyEditTrack.pointsList);
+
+                ctx.currentlyEditTrackDispatch({
+                    type: 'deletePoint',
+                    pointsList: ctx.currentlyEditTrack.pointsList,
+                    layer: ctx.currentlyEditTrack.newRouteLayer,
+                    distance: ctx.currentlyEditTrack.pointsList[ctx.currentlyEditTrack.pointsList.length - 1].dist
```

Comment:
- Distance should be responsibility of Object and never calculated in other places